Map/Map.cpp

```diff
@@ -0,0 +1,358 @@
+}
+    
+void Territory::setNumberOfArmies(int newNumberOfArmies){
+    numberOfArmies = newNumberOfArmies;
+}
+
+void Territory::setContinent(Continent* newContinent){
+    this->continent = newContinent;
+}
+
+//-------------------------- Node class  --------------------------//
+Node::Node(){}
+
+Node::Node(Territory data){
+    Territory* temp = new Territory(data.getTerritoryName(), data.getContinent());
+    this->data = temp;
+}
+
+Node::Node(const Node& original){ //Copy construcotr
+    data = new Territory(*original.data);
+    E = original.E;
+}
+
+//-------------- Deconstructor --------------//
+Node::~Node(){
+    delete[] data;
+}
+
+//-------------- Overloads --------------//
+//overload assignment operator
+Node& Node::operator=(const Node& rhs){
+    if(this != &rhs){
+        *data = *rhs.data;
+        E = rhs.E;
+    }
+
+    return *this;
+}
+
+//-------------- Getters --------------//
+Territory Node::getData(){
+    return *data;
+}
+
+Territory* Node::getDataPtr(){
+    return data;
+}
+
+vector<string> Node::getE(){
+    return E;
+}
+
+//-------------- Setters --------------//
+void Node::setData(Territory* data){
+    this->data = data;
+}
+
+//-------------- Other functions --------------//
+void Node::addEdge(string edge){
+    E.push_back(edge);
+}
+
+
+//-------------------------- Graph class --------------------------//
+//-------------- Constructor --------------//
+Graph::Graph(){}
+
+Graph::Graph(const Graph& original){//Copy constructor
+    //create a deep copy of V
+    for(auto i=0;i<original.getV().size();i++){
+        V.push_back(new Node(*original.V[i]));
+    }
+    
+    //create a deep copy of listOfContinents
+    for(int i=0;i<original.getListOfContinents().size();i++){
+        listOfContinents.push_back(new Continent(*original.listOfContinents[i]));
+    }
+}
+
+//-------------- Deconstructor --------------//
+Graph::~Graph(){
+    for(Node* node : V){
+        delete[] node;
+    }
+
+    for(Continent* continent : listOfContinents){
+        delete[] continent;
+    }
+}
+//-------------- Overloads --------------//
+//overload assignment operator
+Graph& Graph::operator=(const Graph& rhs){
+    if(this != &rhs){
+        //create a deep copy of V
+        for(auto i=0;i<rhs.getV().size();i++){
+            V.push_back(new Node(*rhs.V[i]));
+        }
+        
+        //create a deep copy of listOfContinents
+        for(int i=0;i<rhs.getListOfContinents().size();i++){
+            listOfContinents.push_back(new Continent(*rhs.listOfContinents[i]));
+        }
+    }
+    return *this;
+}
+
+//-------------- Getters --------------//
+vector<Node*> Graph::getV() const{
+    return this->V;
+}
+
+vector<Continent*> Graph::getListOfContinents() const {
+    return this->listOfContinents;
+}
+
+//-------------- Inserting and connecting territories --------------//
+void Graph::insertATerritory(Territory data){
+    Node* newTerritory = new Node(data);
+    V.push_back(newTerritory);
```

Comment:
Why is it in two steps? I think both lines can be combined into one
V.push_back(new Node(data));

Reply:
Has been update accordingly.